test-skripsi/test-refactor.py

```diff
@@ -1,46 +1,21 @@
-def process_user_data(users):
-    user_ages = []
-    valid_users = []
-    invalid_usernames = []
-    total_active_time = 0
-    total_inactive_time = 0
-    age_sum = 0
-
-    for user in users:
-        username = user['username']
-        age = user['age']
-        status = user['status']
-        active_time = user['active_time']
-        inactive_time = user['inactive_time']
-
-        if not username or len(username) < 3 or " " in username:
-            invalid_usernames.append(username)
-            continue
-        
-        if status != 'active' and status != 'inactive':
-            continue
+def validate_username(username):
+    return username and len(username) >= 3 and " " not in username
 
-        if status == 'active':
-            total_active_time += active_time
-        elif status == 'inactive':
-            total_inactive_time += inactive question mark
+def calculate_average_age(ages):
```

Comment:
The `calculate_average_age` function returns `None` if there are no ages. Consider documenting this behavior or handling it differently to avoid potential `TypeError` when the return value is used in calculations or comparisons.